public/app/features/dashboard-scene/settings/variables/VariableEditorForm.tsx

```diff
@@ -19,28 +19,64 @@ import { VariableNameConstraints } from 'app/features/variables/editor/types';
 import { VariableTypeSelect } from './components/VariableTypeSelect';
 import { EditableVariableType, getVariableEditor, hasVariableOptions, isEditableVariableType } from './utils';
 
+const RESERVED_GLOBAL_VARIABLE_NAME_REGEX = /^(?!__).*$/;
+const WORD_CHARACTERS_REGEX = /^\w+$/;
+
 interface VariableEditorFormProps {
   variable: SceneVariable;
   onTypeChange: (type: EditableVariableType) => void;
   onGoBack: () => void;
   onDelete: (variableName: string) => void;
+  isVariableNameTaken: (name: string, key: string | undefined) => boolean;
 }
-
-export function VariableEditorForm({ variable, onTypeChange, onGoBack, onDelete }: VariableEditorFormProps) {
+export function VariableEditorForm({
+  variable,
+  onTypeChange,
+  onGoBack,
+  onDelete,
+  isVariableNameTaken,
+}: VariableEditorFormProps) {
   const styles = useStyles2(getStyles);
-  const { name, type, label, description, hide } = variable.useState();
+  const [nameError, setNameError] = useState<string | null>(null);
+  const { name, type, label, description, hide, key } = variable.useState();
```

Comment:
@dprokop @ivanortegaalba `key`'s definition is `string | undefined` - what are the cases where the `key` value would be `undefined`?